features/telemetry/adapters/nxos/src/main/java/org/opennms/netmgt/telemetry/adapters/nxos/NxosGpbParserUtil.java

```diff
@@ -0,0 +1,213 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.adapters.nxos;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis;
+import org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry;
+
+/**
+ * Utility to parse any key/value metric from gpb messages.
+ *
+ * Since there is no easy way to get key/value metrics from repeated data_gpbkv,
+ * this utility provides some methods to get values for those metrics without
+ * bothering about structure they may follow.
+ *
+ * @author cgorantla
+ */
+public class NxosGpbParserUtil {
+
+    /**
+     * Given a Telemetry Message and field name, get it's value as double.
+     * 
+     * @param   msg  {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry} message
+     * @param   name  field name
+     * @return  value field value
+     */
+    public static Double getValueAsDouble(TelemetryBis.Telemetry msg, String name) {
+        for (TelemetryBis.TelemetryField field : msg.getDataGpbkvList()) {
+            TelemetryBis.TelemetryField subField = findFieldWithName(field, name);
+            if (subField != null) {
+                return getValue(subField);
+            }
+        }
+        return Double.NaN;
+    }
+
+    /**
+     * Given a Telemetry Message and field name, get it's value as string.
+     * 
+     * @param   msg  {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry} message
+     * @param   name  field name
+     * @return  value field value
+     */
+    public static String getValueAsString(TelemetryBis.Telemetry msg, String name) {
+        for (TelemetryBis.TelemetryField field : msg.getDataGpbkvList()) {
+            TelemetryBis.TelemetryField subField = findFieldWithName(field, name);
+            if (subField != null) {
+                return getStringValue(subField);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Given a Telemetry Message and row name, get the list of fields
+     * 
+     * @param   Telemetry message {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry}
+     * @param   name of field
+     * @return  list of Telemetry fields
+     */
+    public static List<TelemetryBis.TelemetryField> getRowsFromTable(Telemetry msg, String name) {
+        Objects.requireNonNull(msg);
```

Comment:
Is there any reason we check for `null` here, but nowhere else?

Reply:
I will remove and will just return null if any of params are null instead of throwing exception